src/PostgREST/ApiRequest.hs

```diff
@@ -125,8 +126,11 @@ userApiRequest schema req reqBody
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
           either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+              Nothing -> Left "All object keys must match"
+              Just json -> Right json)
+            (if BL.null reqBody && isTargetingProc
+              then Right emptyObject
+              else JSON.eitherDecode reqBody)
```

Comment:
I think it is a nice refactoring. How do you feel about extending this to the following:
```
import qualified Data.Either.Utils as EU (maybeToEither)

CTApplicationJSON ->
    (if BL.null reqBody && isTargetingProc
         then Right emptyObject
         else JSON.eitherDecode reqBody)
     >>= EU.maybeToEither "All object keys must match" . ensureUniform . pluralize
CTTextCSV ->
    CSV.decodeByName reqBody
    >>= EU.maybeToEither "All lines must have same number of fields" . ensureUniform . csvToJson
```

?